src/DynamoApplications/StartupUtils.cs

```diff
@@ -228,5 +228,65 @@ public static string SetLocale(CommandLineArguments cmdLineArgs)
             return sb.ToString();
         }
 
+        /// <summary>
+        /// Handler for an assembly load event into a host's appdomain - we need to make sure
+        /// that another addin or package has not loadead another version of a .dll that we require.
+        /// If this happens Dynamo will most likely crash. We should alert the user they
+        /// have an incompatible addin installed.
+        /// 
+        /// </summary>
+        /// <param name="assembly"></param>
+        /// <param name="assemblyNamesToIgnore"></param>
+        /// <returns></returns>
+        public static List<Exception> AppDomainHasMismatchedReferences(Assembly assembly, String[] assemblyNamesToIgnore)
```

Comment:
I agree. Maybe call it GetMismatchedReferencesInAppDomain() or GetMismatchedAppDomainReferences(). It is not clear to me (yet) if this method always is called during startup? E.g. does returning an empty list indicate that everything is OK? You should probably mention this in the comments if that is the case.